File: coding challenge/lookSay.js
// INVERSE LOOK AND SAY

const lookSay = number => {
    let num = String(number)

    let result = ""
    let i = 0

    while (i < num.length) {
        let count = 1
        while ((i + 1 < num.length) && num[i] === num[i+1]){
            i += 1
            count += 1
        }
        result += (String(count) + num[i])
        i += 1
    }

    return result
}

number = 124324

console.log(lookSay(number));

File: look_say.py
def look_say(number: int) -> int:
    num = str(number)

    result = ""
    i = 0

    while i < len(num):
        count = 1
        while (i + 1 < len(num)) and (num[i] == num[i+1]):
            i += 1
            count += 1
        result += (str(count) + num[i])
        i += 1

    return int(result)